test/functional/p2p_invalid_block.py

```diff
@@ -133,5 +139,19 @@ def run_test(self):
         self.log.info("Test inflation by duplicating input")
         peer.send_blocks_and_test([block4], node, success=False,  reject_reason='bad-txns-inputs-duplicate')
 
+        self.log.info("Test accepting identical block after rejecting it due to a future timestamp.")
+        t = int(time.time())
+        # Make block timestamp validity 60 seconds premature to avoid test-timing issues
+        t_premature = 60
+        node.setmocktime(t)
+        block = create_block(tip, create_coinbase(height), t + MAX_FUTURE_BLOCK_TIME + t_premature)
+        block.hashMerkleRoot = block.calc_merkle_root()
+        block.solve()
+        # Need force_send because the block will get rejected without a getdata otherwise
+        peer.send_blocks_and_test([block], node, force_send=True, success=False, reject_reason='time-too-new')
+        node.setmocktime(t + t_premature)
+        peer.send_blocks_and_test([block], node, success=True, timeout=5)
```

Comment:
```suggestion
        peer.send_blocks_and_test([block], node, success=True)
```

No need to lower this. User can adjust via `--timeout-factor=0.1`

Reply:
Thanks. Will update this with or without the time premature (=1) suggestion.

Reply:
Removed in 754e802